src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.1.0.cs

```diff
@@ -26,19 +18,35 @@ public partial class ExpectedExceptionTests : ConditionalWcfTest
     public static void NotExistentHost_Throws_EndpointNotFoundException()
     {
         string nonExistentHost = "http://nonexisthost/WcfService/WindowsCommunicationFoundation";
+        ChannelFactory<IWcfService> factory = null;
+        EndpointAddress endpointAddress = null;
+        BasicHttpBinding binding = null;
+        IWcfService serviceProxy = null;
 
-        BasicHttpBinding binding = new BasicHttpBinding();
-        binding.SendTimeout = TimeSpan.FromMilliseconds(20000);
 
+        // *** VALIDATE *** \\
         EndpointNotFoundException exception = Assert.Throws<EndpointNotFoundException>(() =>
         {
-            using (ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(nonExistentHost)))
+            // *** SETUP *** \\
+            binding = new BasicHttpBinding();
+            binding.SendTimeout = TimeSpan.FromMilliseconds(20000);
+            endpointAddress = new EndpointAddress(nonExistentHost);
+            factory = new ChannelFactory<IWcfService>(binding, endpointAddress);
+            serviceProxy = factory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            try
             {
-                IWcfService serviceProxy = factory.CreateChannel();
-                string response = serviceProxy.Echo("Hello");
+                serviceProxy.Echo("Hello");
+            }
+            finally
+            {
+                // *** ENSURE CLEANUP *** \\
+                ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
             }
         });
 
+        // *** ADDITIONAL VALIDATION FOR NET NATIVE *** \\
         // On .Net Native retail, exception message is stripped to include only parameter
         Assert.True(exception.Message.Contains(nonExistentHost), string.Format("Expected exception message to contain: '{0}'", nonExistentHost));
```

Comment:
>"Expected exception message to contain: '{0}' [](start = 79, length = 45)

Nit: Not blocking. I know this code existed before your changes, but this message would be improved if it said what Exception.Message was, not just what it did not contain.